File: client/JSONFactory.py
'''
Created on Jun 21, 2014

@author: ronaldjosephdesmarais
'''
import json


    
#'{"request":"add_billboard","name":"RONS_test_bb","lat":-142.1111,"lng":48.7777}'
def createInsert(lat,lng,height,name):
    return '{"lat":%s,"lng":%s,"height":%s,"name":"%s"}'%(lat,lng,height,name)

def createDelete(ids):
    return '{"id":%s}'%(ids)

def createGet(ids):
    return '{"id":%s}'%(ids)

def createUpdate(ids,lat,lng,height,name):
    return '{"id":%s,"lat":%s,"lng":%s,"height":%s,"name":"%s"}'%(ids,lat,lng,height,name)

def createSearch(lat=None,lng=None,height=None,lat_rng=None,lng_rng=None,height_rng=None,name=None):
    if name is None:
        return '{"lat":%s,"lng":%s,"height":%s,"lat_rng":%s,"lng_rng":%s,"height_rng":%s}'%(lat,lng,height,lat_rng,lng_rng,height_rng)
    else:
        return '{"name":"%s"}'%name

File: client/java/src/client/JSONFactory.java
package client;

public class JSONFactory {
	public static String createInsert(double lat,double lng,double height,String name){
		return "{\"lat\":"+lat+",\"lng\":"+lng+",\"height\":"+height+",\"name\":\""+name+"\"}";
	}
	
	public static String createUpdate(int id,double lat,double lng,double height,String name){
		return "{\"id\":"+id+",\"lat\":"+lat+",\"lng\":"+lng+",\"height\":"+height+",\"name\":\""+name+"\"}";
	}
	
	public static String createSearch(Double lat, Double lng, Double height, Double latRng, Double lngRng, Double heightRng,String name){
		//return '{"lat":%s,"lng":%s,"height":%s,"lat_rng":%s,"lng_rng":%s,"height_rng":%s}'%(lat,lng,height,lat_rng,lng_rng,height_rng)
	    if (name == null){
	    	return "{\"lat\":"+lat+",\"lng\":"+lng+",\"height\":"+height+",\"lat_rng\":"+latRng+",\"lng_rng\":"+lngRng+",\"height_rng\":"+heightRng+" }";
	    }else{
	    	return "{\"name\":\""+name+"\"}";
	    }
	}
	
	public static String createDelete(int id){
		return "{\"id\":"+id+"}";
	}
	
	public static String createGet(int id){
		return "{\"id\":"+id+"}";
	}
}
